### ISSUE TITLE
initialScrollOffset is incorrect in 0.2.0

### ISSUE THREAD

**Issue by sososdk** [superlistapp/super_sliver_list#35]
Use initialScrollOffset:
```dart
final scrollController = ScrollController(initialScrollOffset: index * 48);
CustomScrollView(
  controller: scrollController,
  ...
);
```

super_sliver_list [0.0.8](https://pub.dev/packages/super_sliver_list/versions/0.0.8) works fine.

super_sliver_list [0.2.0](https://pub.dev/packages/super_sliver_list/versions/0.2.0) initialScrollOffset is incorrect.

**Comment by knopp** [comment]
That is not unexpected. Even during initial layout the may scroll offset gets corrected in some cases. The offset estimation works differently in the lists. In 0.2.0 it is approximation based on existing items.

What is your use-case? Are all your list item fixed-extent? If so, you can provide `extentEstimation`, i.e. 
```dart
SuperSliverList(
   extentEstimation(_,__) => 48,
)
```
which means that estimated item extent will match actual extent and there is no correction.

I think `super_sliver_list` currently needs way to reliably jump to initial index with variable extent lists. That's a missing feature currently.

**Comment by knopp** [comment]
Note that you can get quite far by simply jumping to initial item in first post frame callback. 
```dart
 @override
  void initState() {
    super.initState();
    _scrollController = ScrollController();
    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
      _listControllers.last.jumpToItem(
          index: 50, scrollController: _scrollController, alignment: 0.0);
    });
  }
```

This will show the list initially at the beginning, but only for one frame after which it immediately jump to requested item. The benefit here is that this works with variable extents, you don't need to know the height of your item and they don't need to be same.

**Comment by sososdk** [comment]
`jumpToItem` not work, when `SuperSliverList` wrapped with `MultiSliver`.

Example:
```dart
import 'package:flutter/material.dart';
import 'package:sliver_tools/sliver_tools.dart';
import 'package:super_sliver_list/super_sliver_list.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final scrollController = ScrollController();
  final listController = ListController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: CustomScrollView(
        controller: scrollController,
        slivers: [
          MultiSliver(
            pushPinnedChildren: true,
            children: [
              SliverPinnedHeader(
                child: Container(
                  color: Theme.of(context).colorScheme.inversePrimary,
                  child: const ListTile(title: Text('Group 1')),
                ),
              ),
              SuperSliverList(
                delegate: SliverChildBuilderDelegate(
                  (context, index) => ListTile(title: Text('Item $index')),
                  childCount: 30,
                ),
              )
            ],
          ),
          MultiSliver(
            pushPinnedChildren: true,
            children: [
              SliverPinnedHeader(
                child: Container(
                  color: Theme.of(context).colorScheme.inversePrimary,
                  child: const ListTile(title: Text('Group 2')),
                ),
              ),
              SuperSliverList(
                delegate: SliverChildBuilderDelegate(
                  (context, index) => ListTile(title: Text('Item $index')),
                  childCount: 30,
                ),
              )
            ],
          ),
          MultiSliver(
            pushPinnedChildren: true,
            children: [
              SliverPinnedHeader(
                child: Container(
                  color: Theme.of(context).colorScheme.inversePrimary,
                  child: const ListTile(title: Text('Group 3')),
                ),
              ),
              SuperSliverList(
                listController: listController,
                delegate: SliverChildBuilderDelegate(
                  (context, index) => ListTile(title: Text('Item $index')),
                  childCount: 30,
                ),
              )
            ],
          ),
        ],
      ), //
      floatingActionButton: FloatingActionButton(
        onPressed: () => listController.jumpToItem(
          index: 2,
          scrollController: scrollController,
          alignment: 0.0,
        ),
        tooltip: 'Jump',
        child: const Icon(Icons.gps_fixed),
      ),
    );
  }
}
```

**Comment by sososdk** [comment]
Is there a way to avoid rebound?

https://github.com/superlistapp/super_sliver_list/assets/46806014/53cc1112-4289-405a-9e88-8d4382b88d25

**Comment by knopp** [comment]
The rebound is there because you are showing last item with alignment = 0, which means beginning of viewport. If you want to show last item at the end of the list, you need to set alignment to 1, which will get rid of rebound.

**Comment by sososdk** [comment]
Yes, you are right. 

But I want it to stop sliding when the list may rebound.  Like [scroll_to_index](https://pub.dev/packages/scroll_to_index).

**Comment by knopp** [comment]
That is a good point. I agree current behavior is counter-intuitive. Should be fixed by https://github.com/superlistapp/super_sliver_list/pull/38.

**Comment by knopp** [comment]
Overscroll should be fixed in 0.2.1 (just published).

**Comment by sososdk** [comment]
Thank you so much!
